pkg/service/repair/generator.go

```diff
@@ -158,20 +163,31 @@ func (g *generator) newTableGenerator(keyspace string, tp tablePlan, ring scylla
 	done, allRangesCnt := g.pm.GetCompletedRanges(keyspace, tp.Table)
 	// Always repair unfinished tablet table from scratch as
 	// tablet load balancing is enabled when repair is interrupted.
-	if !g.ringDescriber.IsTabletKeyspace(keyspace) || len(done) == allRangesCnt {
+	tabletKs := g.ringDescriber.IsTabletKeyspace(keyspace)
+	if !tabletKs || len(done) == allRangesCnt {
 		for _, r := range done {
 			delete(todoRanges, r)
 		}
 	}
 
+	var jt jobType
+	switch {
+	case g.plan.SmallTableOptSupport && tp.Small && !tabletKs:
+		jt = optimizeJobType
+	case len(ring.ReplicaTokens) == 1 && tp.Small:
```

Comment:
This is the requirement for using the old small table optimization - we send `min_token:max_token` only for fully replicated and small tables. I'm not sure about the reasoning behind table being fully replicated. I suspect that this is because otherwise SM would be sending `min_token:max_token` for each replica set and this could result in repairing ranges many times instead of just once. On the other hand we could just merge the ranges in a normal way (just concatenate them), but it was never done before.